src/bsd.rs

```diff
@@ -0,0 +1,46 @@
+use std::fs::read_link;
+use std::path::PathBuf;
+use libc::*;
+use std::ptr;
+
+// http://stackoverflow.com/q/799679
+// http://stackoverflow.com/a/1024937
+pub fn get_executable_path() -> Option<PathBuf> {
+    // FreeBSD
+    let mut mib = [0; 4];
+    mib[0] = CTL_KERN;
```

Comment:
Can't this be written as `let mut mib = [ CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 ];`?